src/main/java/io/cdap/plugin/gcp/bigtable/sink/RecordToHBaseMutationTransformer.java

```diff
@@ -41,32 +42,51 @@ public RecordToHBaseMutationTransformer(String keyAlias, String columnFamily) {
 
   public Mutation transform(StructuredRecord record) {
     List<Schema.Field> fields = Objects.requireNonNull(record.getSchema().getFields(), "Schema fields cannot be empty");
-    String key = Objects.requireNonNull(record.get(keyAlias), "Key cannot be null");
-    Put put = new Put(Bytes.toBytes(key));
+    Schema.Field keyField = record.getSchema().getField(keyAlias);
+    if (null == keyField) {
+      throw new UnexpectedFormatException(String.format("Record does not contain key field '%s'. Keys are required",
+                                                        keyAlias));
+    }
+    byte[] rowKeyBytes = convertFieldValueToBytes(record.get(keyAlias), keyField);
+    if (null == rowKeyBytes) {
+      throw new UnexpectedFormatException(String.format("Key field '%s' contained a null value. Keys cannot be null",
+                                                        keyAlias));
+    }
+    Put put = new Put(rowKeyBytes);
     byte[] columnFamilyBytes = Bytes.toBytes(columnFamily);
     for (Schema.Field field : fields) {
       String fieldName = field.getName();
-      try {
-        byte[] valueBytes = convertFieldValueToBytes(record.get(fieldName), field.getSchema());
-        put.addColumn(columnFamilyBytes, Bytes.toBytes(fieldName), valueBytes);
-      } catch (Exception e) {
-        throw new UnexpectedFormatException(String.format("Failed to transform value for field '%s'. Reason: %s",
-                                                          fieldName, e.getMessage()));
-      }
+      byte[] valueBytes = convertFieldValueToBytes(record.get(fieldName), field);
+      put.addColumn(columnFamilyBytes, Bytes.toBytes(fieldName), valueBytes);
     }
     return put;
   }
 
-  private byte[] convertFieldValueToBytes(Object value, Schema fieldSchema) {
+  @Nullable
+  private static byte[] convertFieldValueToBytes(@Nullable Object value, Schema.Field field) {
+    try {
+      return convertFieldValueToBytes(value, field.getSchema());
+    } catch (Exception e) {
+      throw new UnexpectedFormatException(String.format("Failed to transform value for field '%s'. Reason: %s",
+                                                        field.getName(), e.getMessage()));
+    }
+  }
+
+  @Nullable
+  private static byte[] convertFieldValueToBytes(@Nullable Object value, Schema fieldSchema) {
     if (fieldSchema.isNullable()) {
       if (null == value) {
         return null;
       }
       return convertFieldValueToBytes(value, fieldSchema.getNonNullable());
     }
-    switch (fieldSchema.getType()) {
-      case NULL:
+    if (null == value) {
```

Comment:
don't think you need this if block, StructuredRecord.Builder will enforce that nulls cannot be set for a non-nullable field.

Reply:
fixed